<launch>
    <arg name="pkg" default="$(find seddom)" />
    <arg name="method" default="seddom"/>
    <arg name="dataset" default="kitti_15"/>
   
    <arg name="method_config" value="$(arg pkg)/config/methods" />
    <arg name="data_config" value="$(arg pkg)/config/datasets/" />

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find seddom)/rviz/kitti_node.rviz" />
    <node pkg="seddom" type="kitti_node" name="kitti_node" output="screen">

        <param name="dir" value="$(arg pkg)/data/data_$(arg dataset)" />

        <!-- Load method configuration from file -->
        <rosparam file="$(arg method_config)/$(arg method).yaml" command="load" />

        <!-- Load dataset configuration from file -->
        <rosparam file="$(arg data_config)/$(arg dataset).yaml" command="load" />
        
    </node>

</launch>
